src/nitypes/bintime/_absolute_time.py

```diff
@@ -0,0 +1,283 @@
+from __future__ import annotations
+
+import datetime as dt
+from functools import singledispatchmethod
+from typing import Any, ClassVar, SupportsIndex, Union, cast, final, overload
+
+import hightime as ht
+from typing_extensions import Self, TypeAlias
+
+from nitypes._exceptions import invalid_arg_type, invalid_arg_value
+from nitypes.bintime._time_value import (
+    _OTHER_TIME_VALUE_TUPLE,
+    TimeValue,
+    _OtherTimeValue,
+)
+
+_DT_EPOCH_1904 = dt.datetime(1904, 1, 1, tzinfo=dt.timezone.utc)
+_HT_EPOCH_1904 = ht.datetime(1904, 1, 1, tzinfo=dt.timezone.utc)
+
+_OtherAbsoluteTime: TypeAlias = Union[dt.datetime, ht.datetime]
+_OTHER_ABSOLUTE_TIME_TUPLE = (dt.datetime, ht.datetime)
+
+
+@final
+class AbsoluteTime:
+    """An absolute time in NI Binary Time Format (NI-BTF).
+
+    AbsoluteTime represents time as a 128-bit fixed point number with 64-bit whole seconds and
+    64-bit fractional seconds.
+
+    .. warning::
+        The fractional seconds are represented as a binary fraction, which is a sum of inverse
+        powers of 2. Values that are not exactly representable as binary fractions will display
+        rounding error or "bruising" similar to a floating point number.
+
+    AbsoluteTime instances are duck typing compatible with a subset of the method and properties
+    supported by :any:`datetime.datetime` and :any:`hightime.datetime`.
+
+    This class only supports the UTC time zone and does not support timezone-naive times.
```

Comment:
The .NET version supports time zones.

Reply:
We can add it in the future if necessary. For now we can omit it.